main.js

```diff
@@ -225,85 +228,106 @@ function initLeaflet() {
     var leafletcss, leafletjs;
 
     if (legacyLeaflet === true) {
-        leafletcss="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.0-beta.2/leaflet.css";
-        leafletjs="https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.0-beta.2/leaflet.js";
+        leafletcss = 'https://unpkg.com/leaflet@1.0.0-beta.2/dist/leaflet.css';
+        leafletjs = 'https://unpkg.com/leaflet@1.0.0-beta.2/dist/leaflet.js';
+        if (query.debug) {
+          leafletjs = 'https://unpkg.com/leaflet@1.0.0-beta.2/dist/leaflet-src.js';
+        }
     } else {
-        leafletcss="https://unpkg.com/leaflet@1.0.2/dist/leaflet.css";
-        leafletjs="https://unpkg.com/leaflet@1.0.2/dist/leaflet.js";
+        leafletcss = 'https://unpkg.com/leaflet@1.0.3/dist/leaflet.css';
+        leafletjs = 'https://unpkg.com/leaflet@1.0.3/dist/leaflet.js';
+        if (query.debug) {
+          leafletjs = 'https://unpkg.com/leaflet@1.0.3/dist/leaflet-src.js';
+        }
     }
-    document.getElementById("leafletcss").href = leafletcss;
+    document.getElementById('leafletcss').href = leafletcss;
 
     return injectScript(leafletjs);
 }
 
 function initMap() {
-    bugOptions = {}
     window.map = (function () {
-        // console.log('Leaflet version:', window.L.version)
-
         'use strict';
 
-        var map_start_location = [40.70531887544228, -74.00976419448853, 15]; // NYC
+        var map_start_location = {
+          lat: 40.705319,
+          lng: -74.009764,
+          zoom: 15
+        }; // NYC
 
         /*** Map ***/
 
-        // leaflet-style URL hash pattern:
-        // #[zoom],[lat],[lng]
-        var url_hash = window.location.hash.slice(1, window.location.hash.length).split('/');
-
-        if (url_hash.length == 3) {
-            map_start_location = [url_hash[1],url_hash[2], url_hash[0]];
-            // convert from strings
-            map_start_location = map_start_location.map(Number);
+        if (typeof query.lat !== 'undefined' && typeof query.lng !== 'undefined' && typeof query.z !== 'undefined') {
+            map_start_location = {
+                lat: query.lat,
+                lng: query.lng,
+                z: query.z
+            };
+        } else {
+            // leaflet-style URL hash pattern: (legacy)
+            // #[zoom],[lat],[lng]
+            var url_hash = window.location.hash.slice(1, window.location.hash.length).split('/');
+
+            if (url_hash.length == 3) {
+                map_start_location = {
+                    lat: Number(url_hash[1]),
+                    lng: Number(url_hash[2]),
+                    zoom: Number(url_hash[0])
+                };
+            }
```

Comment:
Long ago we made the call for Tangram to use (as somebody requested long ago) a "bog-standard leaflet hash" which would copy and paste easily between Tangram maps and other Leaflet maps… is the query-based style preferable? (Or are we planning to switch everywhere?)